tests/e2e/util.go

```diff
@@ -5748,3 +5748,192 @@ func waitAndGetContainerID(sshClientConfig *ssh.ClientConfig, k8sMasterIP string
+					SecurityContext: fpod.GenerateContainerSecurityContext(isPrivileged),
+				},
+			},
+			RestartPolicy: v1.RestartPolicyOnFailure,
+		},
+	}
+	var volumeMounts = make([]v1.VolumeMount, len(pvclaims))
+	var volumes = make([]v1.Volume, len(pvclaims))
+	for index, pvclaim := range pvclaims {
+		volumename := fmt.Sprintf("volume%v", index+1)
+		volumeMounts[index] = v1.VolumeMount{Name: volumename, MountPath: "/mnt/" + volumename}
+		volumes[index] = v1.Volume{Name: volumename, VolumeSource: v1.VolumeSource{
+			PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{ClaimName: pvclaim.Name, ReadOnly: false}}}
+	}
+	podSpec.Spec.Containers[0].VolumeMounts = volumeMounts
+	podSpec.Spec.Volumes = volumes
+	if nodeSelector != nil {
+		podSpec.Spec.NodeSelector = nodeSelector
+	}
+	podSpec.Labels = podlabels
+	return podSpec
+}
+
+// createVsanDPvcAndPod creates pvc and pod for vsan direct as per wffc policy.
+// It ensures that pvc and pod is in in healthy state.
+func createVsanDPvcAndPod(client clientset.Interface, ctx context.Context,
+	namespace string, sc *storagev1.StorageClass, pvcName string,
+	podName string) (*v1.PersistentVolumeClaim, *v1.Pod) {
+	framework.Logf("Creating pvc %s with storage class %s", pvcName, sc.Name)
+	pvclaim, err := fpv.CreatePVC(client, namespace,
+		getVsanDPersistentVolumeClaimSpecWithStorageClass(namespace,
+			diskSize, sc, pvcName, podName, ""))
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	// Wait for additional 30 seconds to make sure that provision volume claim
+	// remains in pending state waiting for first consumer.
+	ginkgo.By("Waiting for 30 seconds and verifying whether the PVC is still in pending state")
+	time.Sleep(time.Duration(sleepTimeOut) * time.Second)
+
+	ginkgo.By("Expect claim status to be in Pending state")
+	err = fpv.WaitForPersistentVolumeClaimPhase(v1.ClaimPending, client,
+		namespace, pvclaim.Name, framework.Poll, time.Minute)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred(),
+		fmt.Sprintf("Failed to find the volume in pending state with err: %v", err))
+
+	ginkgo.By("Creating a pod")
+	podSpec := getVsanDPodSpec(namespace, nil, []*v1.PersistentVolumeClaim{pvclaim}, false, execCommand, podName)
+	pod, err := client.CoreV1().Pods(namespace).Create(ctx, podSpec, metav1.CreateOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	err = fpod.WaitForPodNameRunningInNamespace(client, pod.Name, namespace)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	ginkgo.By("Expect claim to be in Bound state and provisioning volume passes")
+	err = fpv.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, client,
+		namespace, pvclaim.Name, framework.Poll, time.Minute)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred(), fmt.Sprintf("Failed to provision volume with err: %v", err))
+
+	return pvclaim, pod
+
+}
+
+// assignPolicyToWcpNamespace assigns a set of storage policies to a wcp namespace
+func assignPolicyToWcpNamespace(client clientset.Interface, ctx context.Context,
```

Comment:
limit: The maximum amount of storage (in mebibytes) which can be utilized by the namespace for this specification.
If unset, no limits are placed.

This will assign a set of storage policy to namespace and also set quota for that storagepolicy.
We still need to use create resource quota method to create quota for the namespace.